neuro_san/test/assessor/assessor.py

```diff
@@ -0,0 +1,202 @@
+
+    def add_args(self, arg_parser: argparse.ArgumentParser):
+        """
+        Adds arguments.  Allows subclasses a chance to add their own.
+        :param arg_parser: The argparse.ArgumentParser to add.
+        """
+        arg_parser.add_argument("--test_hocon", type=str,
+                                help="The test case .hocon file to use as a basis for assessment")
+        arg_parser.add_argument("--assessor_agent", type=str, default=None,
+                                help="The assessor agent to use. A default of None implies use of neuro-san stock assess_failure.hocon")
+        arg_parser.add_argument("--connection", default="direct", type=str,
+                                choices=["grpc", "direct", "http", "https"],
+                                help="""
+The type of connection to initiate. Choices are to connect to:
+    "grpc"      - an agent service via gRPC. Needs host and port.
+    "http"      - an agent service via HTTP. Needs host and port.
+    "https"     - an agent service via secure HTTP. Needs host and port.
+    "direct"    - a session via library.
+""")
+        arg_parser.add_argument("--host", type=str, default=None,
+                                help="hostname setting if not running locally")
+        arg_parser.add_argument("--port", type=int, default=AgentSession.DEFAULT_PORT,
+                                help="TCP/IP port to run the Agent gRPC service on")
+
+    def categorize_one_failure(self, fail: Dict[str, Any], failure_modes: List[str]) -> str:
+        """
+        Categorize a single failure instance
+        :param fail: A failure dictionary from asserts.get_fail_dicts()
+        :param failure_modes: A List of strings describing known failure modes
+        :return: A string describing an existing mode of failure or
+                a description of a new mode of failure
+        """
+        session: AgentSession = AgentSessionFactory().create_session(self.args.connection,
+                                                                     self.args.assessor_agent,
+                                                                     hostname=self.args.host,
+                                                                     port=self.args.port)
+
+        text: str = f"""
+The acceptance_criteria is:
+"{fail.get('acceptance_criteria')}".
+
+The text_sample is:
+"{fail.get('text_sample')}".
+
+The known failure_modes are:
+{failure_modes}
+"""
+        input_processor = StreamingInputProcessor(session=session)
+        processor: BasicMessageProcessor = input_processor.get_message_processor()
+        request: Dict[str, Any] = input_processor.formulate_chat_request(text)
+
+        # Call streaming_chat()
+        empty: Dict[str, Any] = {}
+        for chat_response in session.streaming_chat(request):
+            message: Dict[str, Any] = chat_response.get("response", empty)
+            processor.process_message(message, chat_response.get("type"))
```

Comment:
There is perhaps some consolidation to be done here for simple one-shot uses of agents. That's for a later PR